earthquakes/helpers.py

```diff
@@ -0,0 +1,187 @@
+            os.mkdir(self.save_dir)
+            print("Directory ", self.save_dir,  " Created ")
+
+        for i_0, i_f in tqdm(range_save):
+            name = os.path.join(self.save_dir, "{}.pkl".format(i_0))
+            data.loc[i_0:i_f, :].to_pickle(name)
+
+    def get_intervals(self, i_init, window_size=150000):
+        """Funtion to get data from memory
+        """
+        logical_gate = [(i_init in inter) or (i_init + window_size in inter) for inter in self.intervals]
+        get_intervals = [x.left for x in self.intervals[logical_gate]]
+        if len(get_intervals) == 1:
+            get_files = get_intervals
+        else:
+            logical_gate_intervals = (self.index_list >= get_intervals[0]) & (self.index_list <= get_intervals[-1:])
+            get_files = self.index_list[logical_gate_intervals]
+
+        signal = pd.DataFrame([])
+        for file in get_files:
+            new_file = pd.read_pickle(os.path.join(self.save_dir, "{}.pkl".format(file)))
+            signal = pd.concat([signal, new_file], axis=0)
+        return signal.loc[i_init: i_init + window_size - 1, :]
+
+
+def create_feature_dataset_source(caller_cl, feature_computer, xcol="acoustic_data", ycol="time_to_failure",
+                                  stft=False, stft_feature_computer=None, events_id=(245829584, 307838916),
+                                  window_size=150000, step=100):
+    """Samples sequences from the data, computes features for each sequence, and stores the result
+    in a new dataframe. This function is a modification from the 'create feature' at the engineering
+    script.
+
+    Parameters
+    ----------
+    caller_sm: caller class
+    feature_computer: FeatureComputer object or similar,
+        A class that implements a method '.compute()' that takes an array and returns
+        features. It must also have an attribute 'feature_names' that shows the corresponding
+        names of the features.
+    xcol: str, optional (default: "acoustic_data"),
+        The column referring to the the signal data.
+    ycol: str, optional (default: "time_to_failure"),
+        The column referring to the target value.
+    stft: bool, optional (default: False),
+        Whether to calculate the Short Time Fourier Transform.
+    stft_feature_computer: FeatureComputer object or None,
+        The computer for stft features.
+    events_id: tuple
+        index value that define the range of values that we are
+        going to take the data from.
+    window_size: int
+        number of observations to take to get
+        information about y
+    step: int
+        number of observations to skip between ys
+
+    Returns
+    -------
+    feature_data: pd.DataFrame,
+        A new dataframe of shape (number_intervals, number of features) with the new features per sequence.
+        The index corresponds to the y position.
+    """
+    number_intervals = int((events_id[1] - events_id[0])/step)
```

Comment:
True, I will take a look. However, all datasets have been already created with this approximated step.

Reply:
Hi, this should work: 
new_list = list(np.arange(events_id[0], events_id[1], step))
new_list.append(events_id[1])

Instead of:
    number_intervals = int((events_id[1] - events_id[0])/step)
    indices = np.linspace(events_id[0], events_id[1], number_intervals, endpoint = False).astype(int)

Reply:
solved Chris 👍